flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/GenericHiveMetastoreCatalogUtil.java

```diff
@@ -148,14 +161,24 @@ public static CatalogBaseTable createCatalogTable(Table hiveTable) {
 		// Partition keys
 		List<String> partitionKeys = new ArrayList<>();
 
-		if (hiveTable.getPartitionKeys() != null && hiveTable.getPartitionKeys().isEmpty()) {
+		if (hiveTable.getPartitionKeys().isEmpty()) {
 			partitionKeys = hiveTable.getPartitionKeys().stream()
```

Comment:
if hive table partition keys are empty, we we need to iterate thru it?

Reply:
good catch. typo. fixed

Reply:
Keep the null check unless we are sure it will never be null.

Reply:
the partition keys returned from metastore will never be null